R/resultAnnotations.R

```diff
@@ -0,0 +1,925 @@
+    return(fds)
+}
+
+#' @describeIn spliceTypeAnnotations This method annotates the splice event 
+#'     type to junctions in the given results table.
+#' @export
+annotateSpliceEventType <- function(result, txdb, fds, addSpliceType=TRUE, 
+                                    addUTRoverlap=TRUE, BPPARAM=bpparam()){
+    
+    # Create basic annotation of overlap with reference
+    if(!("annotatedJunction" %in% colnames(result))){
+        stop("Column 'annotatedJunction' not found in the results table!\n",
+            "Please run fds <- annotateIntronReferenceOverlap(fds, txdb) ",
+            "first and add it \nto the results table with ",
+            "results(..., additionalColumns = 'annotatedJunction')\n", 
+            "(see examples) before calling this function.")
+    }
+    
+    # convert to data.table if not already
+    if(!is.data.table(result)){
+        result <- as.data.table(result)
+    }
+    
+    # Calculate splice types and frameshift
+    if(isTRUE(addSpliceType)){
+        result <- addSpliceTypeLabels(result, fds, txdb)
+    }
+    
+    # Add UTR labels
+    if(isTRUE(addUTRoverlap)){
+        result <- addUTRLabels(result, txdb)
+    }
+    
+    return(result)
+}
+
+#' @describeIn spliceTypeAnnotations This method flags blacklist regions in 
+#'     the given results table. 
+#' @export
+flagBlacklistRegions <- function(result, blacklist_regions=NULL,
+                                assemblyVersion=c('hg19', 'hg38')){
+    
+    # convert to data.table if not already
+    if(!is.data.table(result)){
+        result <- as.data.table(result)
+    }
+    
+    assemblyVersion <- match.arg(assemblyVersion)
+    if(is.null(blacklist_regions)){
+        blacklist_regions <- 
+            system.file("extdata", "blacklist_regions", 
+                        paste0(assemblyVersion, "-blacklist.v2.bed.gz"), 
+                        package = "FRASER")
+    }
+    if(!file.exists(blacklist_regions)){
+        stop("BED file with blacklist regions does not exist: ", 
+            blacklist_regions)
+    }
+    print("Importing blacklist regions ...")
+    blacklist_gr <- rtracklayer::import(blacklist_regions, format = "BED")
+    result <- addBlacklistLabels(result, blacklist_gr)  
+    return(result)
+}
+
+############# helper functions ##############################
+
+#' blacklist annotation for aberrant splicing events  
+#' @noRd
+addBlacklistLabels <- function(junctions_dt, blacklist_gr){
+    # add the blacklist information
+    print("Set up aberrant splicing granges")
+    colnames(junctions_dt)[which(names(junctions_dt) == "STRAND")] <- "strand2"
+    junctions_gr <- makeGRangesFromDataFrame(junctions_dt)
+    
+    # set to the same seqlevelsstyle
+    seqlevelsStyle(blacklist_gr) <- seqlevelsStyle(junctions_gr)
+    
+    ## create overlap with blacklist and annotate extra column
+    print("find blacklist overlap")
+    black_hits <- unique(from(findOverlaps(junctions_gr, blacklist_gr)))
```

Comment:
As discussed we have to see if this makes sense. I would rather argue it makes only sense if the black region is X bases around the donor or the acceptor. But not if the black region is within the intron but not close to any splice site.

Reply:
thanks for spotting this, I agree, makes more sense to me as well to only check for overlap at the acceptor/donor sites. forgot we wanted to change this over cleaning up the documentation part